assignment.rb

```diff
@@ -0,0 +1,295 @@
+  # Menu must be an array with options starting at 1.
+  def get_menu_selection(menu)
+    men_sel = 0
+    while !(1..menu.count).include?(men_sel)
+      men_sel = get_input(menu.join("\n")).to_i
+      if !(1..menu.count).include?(men_sel)
+        puts "Invalid entry. Try again."
+        puts ""
+      end
+    end
+
+    men_sel
+  end
+
+  # This method takes login info and searches the "database" for a valid
+  # user id and pin to match. If no match is found, valid user is left false.
+  def get_user(login)
+    match_found = false
+    @account_holders.each do |user|
+      if user.name == login[:name] && user.pin == login[:pin]
+        @show_acc_menu = true
+        match_found = true
+        puts "Your account has been verified."
+        puts ""
+        return user
+      end
+    end
+    if match_found == false
+      puts "Something went wrong. Please start over."
+      puts ""
+    end
+  end
+  
+  def display_error
+    puts "Something has gone horribly wrong. This machine"
+    puts "is now set to self detonate. You should run."
+    puts ""
+  end
+
+  # This method is responsible for operating the Account Holder's menu
+  def show_acc_menu
+    # Now display the account menu and get the user's input
+    acc_menu_sel = get_menu_selection(ACC_MENU)
+
+    # Depending on the user's input, perform the correct function
+    case acc_menu_sel
+    # Account Menu option 1 indicates the user wants to check their balance.
+    when 1
+      puts "Your current account balance is: $#{@user.balance}"
+      puts ""
+    # Account Menu option 2 indicates the user would like to withdraw funds
+    when 2
+      # Make a withdrawal
+      withdraw_funds()
+    # Account Menu option 3 indicates the user would like to log out
+    when 3
+      @show_acc_menu = false
+      puts "Returning to the Main Menu."
+      puts ""
+    # If menu options 1-3 have failed, the world might end. Display an error
+    else
+      display_error()
+    end
+  end
+
+  # This method handles the withdrawing funds logic.
+  def withdraw_funds
+    # Ask the user how much they would like to withdraw from their account
+    amount = get_input("How much would you like to withdraw?").to_i
+    # Let's make sure the account and the ATM can handle this transaction
+    if @user.can_withdraw?(amount) && @atm.can_withdraw?(amount)
+      # Passed that check. Now update account amounts in memory and on file...
+      @user.deduct(amount)
+      @atm.deduct(amount)
+      update_data_file(@data_file)
+      # ... and dispense the monies.
+      puts "Sure thing. Dispensing cash below..."
+      puts "" 
+      puts "Your new balance is $#{@user.balance}"
+      puts ""
+    elsif amount <= 0
+      puts "You can not withdraw a negative amount."
+      puts ""
+    elsif !@user.can_withdraw?(amount)
+      puts "You have asked to withdraw more than you have in your account"
+      puts ""
+    elsif !@atm.can_withdraw?(amount)
```

Comment:
lines `182`, `195` and `198` have recurring method calls. It might be a good idea to store these 2 method call return values as variables. Imagine if this was doing some sort of external request--it would call something twice instead of caching it in a variable just once.